File: Cpp/throwns.cpp
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
	int k, c;
	stack<int> a;
	a.push(0);
	cin >> k >> c;
	for(int i = 0; i < c; i++) {
		string next;
		int val;
		cin >> next;
		if(next == "undo") {
			cin >> val;
			while(val--) {
				a.pop();
			}
		} else {
			val = stoi(next);	
			int sum = a.top() + val;
			while(sum < 0) {
				sum += k;
			}
			a.push(sum%k);
		}	
	}	
	cout << a.top() << "\n";
	return 0;
}

File: Java/Throwns.java
import java.util.*;

public class Throwns {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int kids = in.nextInt();
		int com = in.nextInt();
		in.nextLine();
		Stack<Integer> past = new Stack<Integer>();
		int pos = 0;
		for(int i = 0; i < com; i++) {
			String instr = in.next();
			if(instr.equals("undo")) {
				int undos = in.nextInt();
				for(int j = 0; j < undos; j++) {
					int last = past.pop() * -1;
					pos += last;
					while(pos<0) {
						pos += kids;
					}
					pos = pos%kids;
				} 
			} else {
				int next = Integer.parseInt(instr);
				pos += next;
				past.push(next);
				while(pos<0) {
					pos += kids;
				}
				pos = pos%kids;
			}
		}
		System.out.println(pos);
	}
}
